server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPRoute.java

```diff
@@ -0,0 +1,55 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap;
+
+import java.util.function.BiFunction;
+
+import org.reactivestreams.Publisher;
+
+import reactor.netty.http.server.HttpServerRequest;
+import reactor.netty.http.server.HttpServerResponse;
+
+public class JMAPRoute {
+    public interface Action extends BiFunction<HttpServerRequest, HttpServerResponse, Publisher<Void>> {
```

Comment:
it's shorter to have a method defined in the interface than extending a BiFunction type IMO

Reply:
How is it shorter? I'm not completely convinced

Reply:
Nevermind, I think I see

Reply:
No I don't ^^'

Reply:
```
interface Action {
  Publisher<Void> handleRequest(HttpServerRequest request, HttpServerResponse resrponse);
}
```
Ok, it's not exactly shorter but the reason why is for naming important things (like the method name) rather than invoking very generic stuff (BiFunction)